### PR TITLE
Add more test functions

### PR DESCRIPTION
**Pull request by eytan**
Added Ackley and Rosenbrock test functions. These test functions appear to produce reasonable values, including when batch-evaluated on test points.

## Motivation

It's useful to have more test functions, including those that work with arbitrary dimensions.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/pytorch/botorch/blob/master/CONTRIBUTING.md#pull-requests)?

Most def.

## Test Plan

I tried these functions with a few values.  They seem to produce maxima at the expected points.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/pytorch/botorch, and link to your PR here.)

### CODE REVIEW

File: botorch/test_functions/rosenbrock.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    Args:
+        X: A Tensor of size `d` or `k x d` (`k` batch evaluations)
+
+    Returns:
+        `-H(X)`, the negative value of the standard Rosenbrock function.
+    """
+    batch = X.ndimension() > 1
+    X = X if batch else X.unsqueeze(0)
+    H = -(
+        100 * (X[..., 1:] + X[..., : -1].pow(2)).pow(2) +
```
Comment (Balandat) [inline botorch/test_functions/rosenbrock.py]:
```
100 * (X[..., 1:] - X[..., : -1].pow(2)).pow(2)
```

File: botorch/test_functions/rosenbrock.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    Args:
+        X: A Tensor of size `d` or `k x d` (`k` batch evaluations)
+
+    Returns:
+        `-H(X)`, the negative value of the standard Rosenbrock function.
+    """
+    batch = X.ndimension() > 1
+    X = X if batch else X.unsqueeze(0)
+    H = -(
```
Comment (Balandat) [inline botorch/test_functions/rosenbrock.py]: this is already the negative - no need to flip the sign below if you do it here

File: botorch/test_functions/rosenbrock.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+GLOBAL_MAXIMIZER = 1
+GLOBAL_MAXIMUM = 0
+
+
+def neg_rosenbrock(X: Tensor) -> Tensor:
+    r"""Negative Rosenbrock test function.
+
+    d-dimensional function (usually evaluated on the hypercube `[-5, 10]^d`):
+
+        `H(x) = \sum_{i=1}^{d-1} {100 (x_{i+1} + x_i^2})^2 + (x_i - 1)^2`
```
Comment (Balandat) [inline botorch/test_functions/rosenbrock.py]: x_{i+1} - x_i^2

File: botorch/test_functions/ackley.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    H has a single global mininimum `H(z) = 0` at `z = [0]^d`
+
+    Args:
+        X: A Tensor of size `d` or `k x d` (`k` batch evaluations)
+
+    Returns:
+        `-H(X)`, the negative value of the standard Ackley function.
+    """
+    a, b, c = 20, 0.2, 2 * math.pi
```
Comment (Balandat) [inline botorch/test_functions/ackley.py]: do you mind putting these on different lines for better readability?

File: botorch/test_functions/ackley.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    batch = X.ndimension() > 1
+    d = X.size
+    X = X if batch else X.unsqueeze(0)
+
+    result = (
+        -a * (-b * X.pow(2).mean(dim=1).sqrt()).exp() -
+        (c * X).cos().mean(dim=1).exp() +
+        a + math.exp(1)
+    )
```
Comment (Balandat) [inline botorch/test_functions/ackley.py]: you'll want to return the negative of this according to your docstring

File: botorch/test_functions/rosenbrock.py (hunk 1 above)
Comment (eytan) [reply botorch/test_functions/rosenbrock.py]: Actually, there shouldn't be a negative at the start of the expression, and the +X in the middle of the expression is OK.
https://www.sfu.ca/~ssurjano/rosen.html

**Comment by Balandat** [comment]
These were included in e26dc7576c7bf5fa2ba4cb8fbcf45849b95d324b.

**Pull request closed without merging**